lang/cs/Org.Apache.REEF.Network/Group/Driver/Impl/WritableGeneralGroupCommunicationMessage.cs

```diff
@@ -0,0 +1,127 @@
+﻿/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using System;
+using System.ComponentModel;
+using System.IO;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Org.Apache.REEF.Network.Group.Codec;
+using System.Runtime.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Hadoop.Avro;
+using Org.Apache.REEF.Network.StreamingCodec;
+using Org.Apache.REEF.Wake.Remote;
+using System.Threading.Tasks;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Tang.Interface;
+
+
+namespace Org.Apache.REEF.Network.Group.Driver.Impl
+{
+    /// <summary>
+    /// Messages sent by MPI Operators. This is the abstract class inherited by 
+    /// WritableGroupCommunicationMessage but seen by Network Service
+    /// </summary>
+   [Obsolete("Need to remove Iwritable and use IstreamingCodec. Please see Jira REEF-295 ", false)]
+    public abstract class WritableGeneralGroupCommunicationMessage : IWritable
```

Comment:
WritableGroupCommunicationMessage  inherits from WritableGeneralGroupCommunicationMessage. My question was why introduce two layers instead of combine those two classes. Any possibility we will have another subclass that will inherit from WritableGeneralGroupCommunicationMessage in the future?